### ISSUE TITLE
Second Jupyter notebook creation always gets stuck

### ISSUE THREAD

**Issue by nobuto-m** [canonical/bundle-kubeflow#460]
jupyter-controller gives an error `controller-runtime.controller   Reconciler error        {"controller": "notebook", "request": "admin/second-notebook", "error": "Notebook.kubeflow.org \"second-notebook\" is invalid: status.conditions: Invalid value: \"null\": status.conditions in body must be of type array: \"null\""}` and the following trace on the second notebook creation.

```
2022-06-06T10:16:48.924Z        INFO    controllers.Notebook    Updating StatefulSet    {"notebook": "admin/second-notebook", "namespace": "admin", "name": "second-notebook"}
2022-06-06T10:16:48.943Z        INFO    controllers.Notebook    Updating Status {"notebook": "admin/second-notebook", "namespace": "admin", "name": "second-notebook"}
2022-06-06T10:16:48.949Z        ERROR   controller-runtime.controller   Reconciler error        {"controller": "notebook", "request": "admin/second-notebook", "error": "Notebook.kubeflow.org \"second-notebook\" is invalid: status.conditions: Invalid value: \"null\": status.conditions in body must be of type array: \"null\""}
github.com/go-logr/zapr.(*zapLogger).Error
        /go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:218
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:192
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:171
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88
```

How to reproduce:

1. deploy kubeflow on microk8s
```
sudo snap install microk8s --classic --channel 1.21

microk8s status --wait-ready
microk8s enable dns storage ingress metallb:10.64.140.43-10.64.140.49
microk8s kubectl wait pod --all -A --for condition=Ready

juju bootstrap microk8s
juju add-model kubeflow
juju deploy --trust kubeflow
microk8s kubectl wait -n kubeflow pod --all --for condition=Ready

juju config dex-auth public-url=http://10.64.140.43.nip.io
juju config oidc-gatekeeper public-url=http://10.64.140.43.nip.io
juju config dex-auth static-username=admin
juju config dex-auth static-password=admin
microk8s kubectl wait -n kubeflow pod --all --for condition=Ready
```
2. access to Kubeflow WebUI
3. create a Jupyter notebook as "first-notebook" with the default configuration
4. create another notebook as "second-notebook" with the same configuration
5. the second one is stuck at waiting to be available in the UI

Expected:
the second one gets active within a few minutes

Workaround:
Press the stop button on the note book creation and press the play button to recommence the creation.

![image](https://user-images.githubusercontent.com/4356209/172142026-2fb798a3-d251-4e17-b363-70c35834462e.png)
 
I don't see any difference from the log of `pod/first-notebook-0` and `pod/second-notebook-0`, and both pods are running.

```
$ microk8s kubectl get pod -n admin
NAME                                               READY   STATUS    RESTARTS   AGE
ml-pipeline-visualizationserver-569ccd5d86-nql9p   1/1     Running   0          65m
ml-pipeline-ui-artifact-77dfb58d8b-9njcd           1/1     Running   0          65m
first-notebook-0                                   1/1     Running   0          15m
second-notebook-0                                  1/1     Running   0          14m
```

[microk8s kubectl logs -n admin pod/first-notebook-0]
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/01-copy-tmp-home: Operation not permitted
s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/jupyterlab/run: Operation not permitted
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-copy-tmp-home: executing... 
[cont-init.d] 01-copy-tmp-home: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
/opt/conda/lib/python3.8/site-packages/jupyter_server/transutils.py:13: FutureWarning: The alias `_()` will be deprecated. Use `_i18n()` instead.
  warnings.warn(warn_msg, FutureWarning)
[I 2022-06-06 10:05:02.719 ServerApp] jupyter_server_mathjax | extension was successfully linked.
[I 2022-06-06 10:05:02.728 ServerApp] jupyterlab | extension was successfully linked.
[I 2022-06-06 10:05:02.728 ServerApp] jupyterlab_git | extension was successfully linked.
[I 2022-06-06 10:05:02.736 ServerApp] Writing notebook server cookie secret to /home/jovyan/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2022-06-06 10:05:03.636 ServerApp] nbclassic | extension was successfully linked.
[I 2022-06-06 10:05:03.636 ServerApp] nbdime | extension was successfully linked.
[W 2022-06-06 10:05:03.717 ServerApp] All authentication is disabled.  Anyone who can connect to this server will be able to run code.
[I 2022-06-06 10:05:03.722 ServerApp] nbclassic | extension was successfully loaded.
[I 2022-06-06 10:05:03.723 ServerApp] jupyter_server_mathjax | extension was successfully loaded.
[I 2022-06-06 10:05:03.724 LabApp] JupyterLab extension loaded from /opt/conda/lib/python3.8/site-packages/jupyterlab
[I 2022-06-06 10:05:03.724 LabApp] JupyterLab application directory is /opt/conda/share/jupyter/lab
[I 2022-06-06 10:05:03.727 ServerApp] jupyterlab | extension was successfully loaded.
[I 2022-06-06 10:05:03.729 ServerApp] jupyterlab_git | extension was successfully loaded.
[I 2022-06-06 10:05:03.952 ServerApp] nbdime | extension was successfully loaded.
[I 2022-06-06 10:05:03.995 ServerApp] Serving notebooks from local directory: /home/jovyan
[I 2022-06-06 10:05:03.995 ServerApp] Jupyter Server 1.8.0 is running at:
[I 2022-06-06 10:05:03.995 ServerApp] http://first-notebook-0:8888/notebook/admin/first-notebook/lab
[I 2022-06-06 10:05:03.995 ServerApp]     http://127.0.0.1:8888/notebook/admin/first-notebook/lab
[I 2022-06-06 10:05:03.995 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```

[microk8s kubectl logs -n admin pod/second-notebook-0]
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/01-copy-tmp-home: Operation not permitted
s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/jupyterlab/run: Operation not permitted
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-copy-tmp-home: executing... 
[cont-init.d] 01-copy-tmp-home: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
/opt/conda/lib/python3.8/site-packages/jupyter_server/transutils.py:13: FutureWarning: The alias `_()` will be deprecated. Use `_i18n()` instead.
  warnings.warn(warn_msg, FutureWarning)
[I 2022-06-06 10:05:39.836 ServerApp] jupyter_server_mathjax | extension was successfully linked.
[I 2022-06-06 10:05:39.844 ServerApp] jupyterlab | extension was successfully linked.
[I 2022-06-06 10:05:39.844 ServerApp] jupyterlab_git | extension was successfully linked.
[I 2022-06-06 10:05:39.851 ServerApp] Writing notebook server cookie secret to /home/jovyan/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2022-06-06 10:05:40.740 ServerApp] nbclassic | extension was successfully linked.
[I 2022-06-06 10:05:40.740 ServerApp] nbdime | extension was successfully linked.
[W 2022-06-06 10:05:40.826 ServerApp] All authentication is disabled.  Anyone who can connect to this server will be able to run code.
[I 2022-06-06 10:05:40.832 ServerApp] nbclassic | extension was successfully loaded.
[I 2022-06-06 10:05:40.832 ServerApp] jupyter_server_mathjax | extension was successfully loaded.
[I 2022-06-06 10:05:40.833 LabApp] JupyterLab extension loaded from /opt/conda/lib/python3.8/site-packages/jupyterlab
[I 2022-06-06 10:05:40.833 LabApp] JupyterLab application directory is /opt/conda/share/jupyter/lab
[I 2022-06-06 10:05:40.836 ServerApp] jupyterlab | extension was successfully loaded.
[I 2022-06-06 10:05:40.839 ServerApp] jupyterlab_git | extension was successfully loaded.
[I 2022-06-06 10:05:41.103 ServerApp] nbdime | extension was successfully loaded.
[I 2022-06-06 10:05:41.104 ServerApp] Serving notebooks from local directory: /home/jovyan
[I 2022-06-06 10:05:41.104 ServerApp] Jupyter Server 1.8.0 is running at:
[I 2022-06-06 10:05:41.104 ServerApp] http://second-notebook-0:8888/notebook/admin/second-notebook/lab
[I 2022-06-06 10:05:41.104 ServerApp]     http://127.0.0.1:8888/notebook/admin/second-notebook/lab
[I 2022-06-06 10:05:41.104 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
```

**Comment by nobuto-m** [comment]
[deployment_jupyter-controller.log](https://github.com/canonical/bundle-kubeflow/files/8843736/deployment_jupyter-controller.log)

**Comment by DomFleischmann** [comment]
This is a known upstream [bug](https://github.com/kubeflow/kubeflow/issues/6056) we are trying to get it fixed for the next release.

**Comment by nobuto-m** [comment]
> This is a known upstream [bug](https://github.com/kubeflow/kubeflow/issues/6056) we are trying to get it fixed for the next release.

Thank you for the pointer. That makes sense.

What's a possible fix? Is it going to be this one? https://github.com/kubeflow/kubeflow/pull/6415 If so it looks like the review is pending for a while.

**Comment by kimwnasptd** [comment]
@apo-ger let's also follow these instructions to try and reproduce the issue with the `nil` conditions

**Comment by camille-rodriguez** [comment]
I hit that as well - the pod is running but the notebook is still in status "blue-circle" and doesn't allow for a connection (see image). I've been able to hit stop, start and then it lets me connect to it . 
![image](https://user-images.githubusercontent.com/42066843/186990156-e1eec149-8854-44f1-b2b9-c43096450755.png)

Pod logs (looks good)
`ubuntu@ip-172-31-7-175:~$ kubectl logs test-rolling-status-0 -n admin
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
s6-chown: fatal: unable to chown /var/run/s6/etc/cont-init.d/01-copy-tmp-home: Operation not permitted
s6-chown: fatal: unable to chown /var/run/s6/etc/services.d/jupyterlab/run: Operation not permitted
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 01-copy-tmp-home: executing... 
[cont-init.d] 01-copy-tmp-home: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
/opt/conda/lib/python3.8/site-packages/jupyter_server/transutils.py:13: FutureWarning: The alias `_()` will be deprecated. Use `_i18n()` instead.
  warnings.warn(warn_msg, FutureWarning)
[I 2022-08-26 15:31:02.684 ServerApp] jupyter_server_mathjax | extension was successfully linked.
[I 2022-08-26 15:31:02.737 ServerApp] jupyterlab | extension was successfully linked.
[I 2022-08-26 15:31:02.737 ServerApp] jupyterlab_git | extension was successfully linked.
[I 2022-08-26 15:31:02.749 ServerApp] Writing notebook server cookie secret to /home/jovyan/.local/share/jupyter/runtime/jupyter_cookie_secret
[I 2022-08-26 15:31:03.949 ServerApp] nbclassic | extension was successfully linked.
[I 2022-08-26 15:31:03.950 ServerApp] nbdime | extension was successfully linked.
[W 2022-08-26 15:31:04.043 ServerApp] All authentication is disabled.  Anyone who can connect to this server will be able to run code.
[I 2022-08-26 15:31:04.052 ServerApp] nbclassic | extension was successfully loaded.
[I 2022-08-26 15:31:04.053 ServerApp] jupyter_server_mathjax | extension was successfully loaded.
[I 2022-08-26 15:31:04.054 LabApp] JupyterLab extension loaded from /opt/conda/lib/python3.8/site-packages/jupyterlab
[I 2022-08-26 15:31:04.054 LabApp] JupyterLab application directory is /opt/conda/share/jupyter/lab
[I 2022-08-26 15:31:04.059 ServerApp] jupyterlab | extension was successfully loaded.
[I 2022-08-26 15:31:04.063 ServerApp] jupyterlab_git | extension was successfully loaded.
[I 2022-08-26 15:31:04.365 ServerApp] nbdime | extension was successfully loaded.
[I 2022-08-26 15:31:04.366 ServerApp] Serving notebooks from local directory: /home/jovyan
[I 2022-08-26 15:31:04.366 ServerApp] Jupyter Server 1.8.0 is running at:
[I 2022-08-26 15:31:04.366 ServerApp] http://test-rolling-status-0:8888/notebook/admin/test-rolling-status/lab
[I 2022-08-26 15:31:04.366 ServerApp]     http://127.0.0.1:8888/notebook/admin/test-rolling-status/lab
[I 2022-08-26 15:31:04.366 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
`
It does seem like the same issue - "invalid value \"null\" is seen in the jupyter controller pod logs.
`2022-08-26T20:53:07.202Z	INFO	controllers.Notebook	Updating Status	{"notebook": "admin/test-rolling-status", "namespace": "admin", "name": "test-rolling-status"}
2022-08-26T20:53:07.209Z	ERROR	controller-runtime.controller	Reconciler error	{"controller": "notebook", "request": "admin/test-rolling-status", "error": "Notebook.kubeflow.org \"test-rolling-status\" is invalid: status.conditions: Invalid value: \"null\": status.conditions in body must be of type array: \"null\""}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:218
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:192
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:171
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
	/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88
2022-08-26T20:53:17.810Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:53:17.827Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:54:17.849Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:54:17.864Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:55:17.875Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:55:17.901Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:56:17.914Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:56:17.929Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:57:17.940Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:57:17.955Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:58:17.966Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:58:17.980Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:59:17.991Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T20:59:18.007Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T21:00:18.017Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T21:00:18.032Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T21:01:18.042Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T21:01:18.056Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T21:02:18.067Z	INFO	controllers.Notebook	Updating StatefulSet	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
2022-08-26T21:02:18.080Z	INFO	controllers.Notebook	Updating Notebook CR state: 	{"notebook": "admin/test2-minio", "namespace": "admin", "name": "test2-minio"}
`

**Comment by apo-ger** [comment]
@camille-rodriguez , @nobuto-m  Check this [PR](https://github.com/kubeflow/kubeflow/pull/6628#) which resolves these errors

**Comment by DomFleischmann** [comment]
This is now fixed in Kubeflow 1.6 both upstream and on our distribution. I would encourage anyone who is still hitting this to upgrade to it.
